### PR TITLE
Workspaces and Resource dropdowns are FormSelect

### PR DESCRIPTION
**Pull request by andrewballantyne**
https://issues.redhat.com/browse/ODC-3741

Continuation of https://github.com/openshift/console/pull/8595 as requested by @bgliwa01 

/cc @bgliwa01 @karthikjeeyar 
cc @serenamarie125 

pb-sidebar-form-select

### CODE REVIEW

**Comment by andrewballantyne** [comment]
cc @makambalaji This covers your dropdown problem in the sidebar fwiw.

**Review by bgliwa01 — APPROVED** [review]
lgtm! Thank you Andrew!

**Comment by debsmita1** [comment]
Looks like you have missed using FormSelect for the workspaces dropdown in the pipeline start modal

Screenshot from 2021-04-15 12-40-50

**Comment by andrewballantyne** [comment]
> Looks like you have missed using FormSelect for the workspaces dropdown in the pipeline start modal
> 
> Screenshot from 2021-04-15 12-40-50

@bgliwa01 Thoughts? We don't want to convert all dropdowns in the Console to be this state... and for ConfigMap/Secret/PVC the next dropdown will need a search field so it cannot be converted. Where is the line we draw? My thought was it more-so for Pipeline Builder more than anything else. Do we want to expand the use-cases?

cc @serenamarie125

**Comment by bgliwa01** [comment]
> > Looks like you have missed using FormSelect for the workspaces dropdown in the pipeline start modal
> > Screenshot from 2021-04-15 12-40-50
> 
> @bgliwa01 Thoughts? We don't want to convert all dropdowns in the Console to be this state... and for ConfigMap/Secret/PVC the next dropdown will need a search field so it cannot be converted. Where is the line we draw? My thought was it more-so for Pipeline Builder more than anything else. Do we want to expand the use-cases?
> 
> cc @serenamarie125

I think for now keep it just the Pipeline builder.

**Comment by debsmita1** [comment]
@andrewballantyne  Tried creating a pipeline using one of the samples, the placeholder is missing if I remove the input resource
pb

**Comment by andrewballantyne** [comment]
Good catch, I'll look into that.

**Comment by andrewballantyne** [comment]
> @andrewballantyne Tried creating a pipeline using one of the samples, the placeholder is missing if I remove the input resource
> pb

PF Bug: https://github.com/patternfly/patternfly-react/issues/5687

Put in a workaround for the time being. Thanks for catching that.

File: frontend/packages/console-shared/src/components/formik-fields/field-types.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ export interface DropdownFieldProps extends FieldProps {
   onChange?: (value: string) => void;
 }
 
-export type FormSelectFieldOptions<T = any> = {
+export type FormSelectFieldOption<T = any> = {
```
Comment (andrewballantyne) [inline frontend/packages/console-shared/src/components/formik-fields/field-types.ts:66]: Oops... Option types are singular not plural 🤦

**Comment by andrewballantyne** [comment]
fixed a mistake I made where dropdowns didn't get disabled 🤦

File: frontend/packages/console-shared/src/components/formik-fields/field-types.ts [hunk 2]
```diff
@@ -63,16 +63,16 @@ export interface DropdownFieldProps extends FieldProps {
...
   label: string;
   value: T;
   isPlaceholder?: boolean;
   isDisabled?: boolean;
 };
 
 export type FormSelectFieldProps = FieldProps & {
-  disabled?: boolean;
-  options: FormSelectFieldOptions[];
+  isDisabled?: boolean;
```
Comment (andrewballantyne) [inline frontend/packages/console-shared/src/components/formik-fields/field-types.ts:74]: This is the disable dropdown fix.

**Comment by debsmita1** [comment]
It shows an empty workspace dropdown when there are no workspaces
Screenshot from 2021-04-19 12-38-56

The input resources dropdown works fine
Screenshot from 2021-04-19 13-00-31

**Comment by andrewballantyne** [comment]
Doh, that's unrelated to this PR... but a valid issue... We should trim empty items 😓

**Comment by andrewballantyne** [comment]
@debsmita1 check the diff on the 2nd force push for the fix to empty workspaces showing up in the dropdown.

**Comment by debsmita1** [comment]
Thanks @andrewballantyne !
verified the changes, and it works fine now
/lgtm

**Pull request merged**